packages/black-forest-labs/src/black-forest-labs-image-model.ts

```diff
@@ -0,0 +1,370 @@
+      warnings,
+      providerMetadata: {
+        blackForestLabs: {
+          images: [
+            {
+              ...(resultSeed != null && { seed: resultSeed }),
+              ...(resultStartTime != null && { start_time: resultStartTime }),
+              ...(resultEndTime != null && { end_time: resultEndTime }),
+              ...(resultDuration != null && { duration: resultDuration }),
+            },
+          ],
+        },
+      },
+      response: {
+        modelId: this.modelId,
+        timestamp: currentDate,
+        headers: responseHeaders,
+      },
+    };
+  }
+
+  private async pollForImageUrl({
+    pollUrl,
+    requestId,
+    headers,
+    abortSignal,
+  }: {
+    pollUrl: string;
+    requestId: string;
+    headers: Record<string, string | undefined>;
+    abortSignal: AbortSignal | undefined;
+  }): Promise<{
+    imageUrl: string;
+    seed?: number;
+    start_time?: number;
+    end_time?: number;
+    duration?: number;
+  }> {
+    const url = new URL(pollUrl);
+    if (!url.searchParams.has('id')) {
+      url.searchParams.set('id', requestId);
+    }
+
+    for (let i = 0; i < DEFAULT_MAX_POLL_ATTEMPTS; i++) {
+      const { value } = await getFromApi({
+        url: url.toString(),
+        headers,
+        failedResponseHandler: bflFailedResponseHandler,
+        successfulResponseHandler: createJsonResponseHandler(bflPollSchema),
+        abortSignal,
+        fetch: this.config.fetch,
+      });
+
+      const status = value.status;
+      if (status === 'Ready') {
+        if (typeof value.result?.sample === 'string') {
+          return {
+            imageUrl: value.result.sample,
+            seed: value.result.seed ?? undefined,
+            start_time: value.result.start_time ?? undefined,
+            end_time: value.result.end_time ?? undefined,
+            duration: value.result.duration ?? undefined,
+          };
+        }
+        throw new Error(
+          'Black Forest Labs poll response is Ready but missing result.sample',
+        );
+      }
+      if (status === 'Error' || status === 'Failed') {
+        throw new Error('Black Forest Labs generation failed.');
+      }
+
+      await delay(DEFAULT_POLL_INTERVAL_MILLIS);
+    }
+
+    throw new Error('Black Forest Labs generation timed out.');
+  }
+}
+
+export const blackForestLabsImageProviderOptionsSchema = lazySchema(() =>
+  zodSchema(
+    z.object({
+      seed: z.number().int().nullish(),
+      prompt_upsampling: z.boolean().nullish(),
+      safety_tolerance: z.number().int().min(0).max(6).nullish(),
+      output_format: z.enum(['jpeg', 'png']).nullish(),
+      webhook_url: z.url().nullish(),
+      webhook_secret: z.string().nullish(),
+      input_image: z.string().nullish(),
+      width: z.number().int().nullish(),
+      height: z.number().int().nullish(),
+      image_prompt: z.string().nullish(),
+      image_prompt_strength: z.number().min(0).max(1).nullish(),
+      raw: z.boolean().nullish(),
```

Comment:
the AI SDK uses camelCase for options, we then match it to the actual API arguments in the model code, so `prompt_upsampling` should be `promptUpsampling` etc